### DISCUSSION TITLE
WS disconnected after Node-Red Deploy

### DISCUSSION THREAD

**Discussion by CommanderB** [ybizeul/StreamDeckWS#12] [Show and tell]
Note that when you deploy changes in Node-Red, sometimes the websocket connection gets lost. Anything from StreamDeckWS appears to black hole.

There is a simple fix - on StreamDeck flip to a different profile then back. 

I'm guessing the willAppear event forces the reconnect.

**Comment by ybizeul** [comment]
I usually use partial redeploy, node entire flows :

![image](https://user-images.githubusercontent.com/182035/104818094-b0500300-5825-11eb-8eb4-00435d62f290.png)

Maybe I should take a look into this and see if there is a way to implement the reconnect from StreamDeckWS side.

**Reply by CommanderB** [reply]
Thanks. I hadnt thought of that! I will try and see if it avoids the problem
